snapcraft/internal/project_loader/_config.py

```diff
@@ -245,23 +246,31 @@ def _ensure_no_duplicate_app_aliases(self):
         if duplicates:
             raise errors.DuplicateAliasError(aliases=duplicates)
 
-    def install_package_repositories(self) -> None:
-        keys_path = self.project._get_keys_path()
+    def _get_required_package_repositories(self) -> List[PackageRepository]:
+        package_repos = self.project._snap_meta.package_repositories.copy()
 
-        # Install repositories configured by 'package-repositories'.
-        changes = [
-            package_repo.install(keys_path=keys_path)
-            for package_repo in self.project._snap_meta.package_repositories
+        v1_plugins = [
+            part.plugin
+            for part in self.all_parts
+            if isinstance(part.plugin, plugins.v1.PluginV1)
         ]
+        for plugin in v1_plugins:
+            package_repos.extend(plugin.get_required_package_repositories())
```

Comment:
```suggestion
        package_repos += [p.get_required_package_repositories() for p in v1_plugins]
```

Reply:
I do believe that would make it a list of lists...?

Reply:
ah. get_required_package_repositories returns a list, then yeah, not ideal